src/components/Facility/FacilityForm.tsx

```diff
@@ -193,7 +202,11 @@ export default function FacilityForm(props: FacilityProps) {
 
   useEffect(() => {
     if (facilityId) return;
+    const orgLevels = extractHierarchyLevels(org);
+    const districtMatch =
+      districtOrg && orgLevels.some((level) => level.name === districtOrg.name);
     const levels: Organization[] = [];
+    if (districtMatch) return;
     if (stateOrg) levels.push(stateOrg);
     if (districtOrg) levels.push(districtOrg);
     if (!stateOrg && !districtOrg && org) levels.push(org);
```

Comment:
let's keep the namings agnostic of state/district/etc. some countries use different namings altogether.